### PR TITLE
Enable test coverage on azure pipeline

### PR DESCRIPTION
**Pull request by kcudnik**

### CODE REVIEW

File: .azure-pipelines/build-template.yml [hunk 1]
```diff
@@ -97,20 +101,48 @@ jobs:
...
   - script: |
       sudo cp azsyslog.conf /etc/rsyslog.conf
       sudo service rsyslog restart
     displayName: "Update rsyslog.conf"
   - ${{ if eq(parameters.run_unit_test, true) }}:
     - script: |
+        set -ex
+        git clone https://github.com/Spacetown/gcovr.git
+        cd gcovr/
+        git checkout origin/recursive_search_file
```
Comment (xumia) [inline .azure-pipelines/build-template.yml:117]: @kcudnik , looks like the branch recursive_search_file is removed, can we change to the other branch?

Comment (kcudnik) [reply .azure-pipelines/build-template.yml:117]: please update template

**Pull request merged**